### PR TITLE
Chore - add helper script to configure a new testnet parachain

### PR DESCRIPTION
**Pull request by ns212**

### CODE REVIEW

File: scripts/init-testnet-parachain.sh [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    polkadot-js-api --ws "${PARACHAIN_WSS}" --sudo --seed "${PARACHAIN_SEED}" \
+        tx.tokens.setBalance \
+        5DqzGaydetDXGya818gyuHA7GAjEWRsQN6UWNKpvfgq2KyM7 \
+        "{ \"Token\": ${TOKEN_IDS[$T]} }" \
+        20000000000000 \
+        0
+  done
+}
+
+function setup_lending_markets {
```
Comment (nud3l) [inline scripts/init-testnet-parachain.sh:88]: Lending markets are different between interlay and kintsugi testnets. So there needs to be a gate somewhere that sets up different markets.

Interlay: DOT, IBTC, USDT, INTR. Also note that DOT, and INTR have just 10 decimals.

Kintsugi: KSM, KBTC, USDT, KINT.

File: scripts/init-testnet-parachain.sh [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+      1
+}
+
+function setup_vault_registry {
+  # TODO
+}
+
+function fund_faucet_account {
+  # Fund the faucet accounts with tokens
+  TOKENS=( "KSM" "KINT" )
```
Comment (nud3l) [inline scripts/init-testnet-parachain.sh:76]: Need to add USDT and KBTC for Kintsugi.

For Interlay we need DOT, IBTC, USDT and INTR.

**Review by nud3l — COMMENTED** [review]
Maybe we should move this script to the scripts here: https://github.com/interlay/interbtc-api/tree/master/scripts ?

**Comment by sander2** [comment]
> Maybe we should move this script to the scripts here: https://github.com/interlay/interbtc-api/tree/master/scripts ?

If this were a js script I would have agreed, but for this bash script I think this repo is fine as well

**Comment by sander2** [comment]
One missing step is to set the current clients version (used by the client auto-updater). We need to make a call like `clientsInfo.setCurrentClientRelease(vault, "https://github.com/interlay/interbtc-clients/releases/download/1.17.6/vault-parachain-metadata-kintsugi-testnet", 0x8393f5064fc88b2486014bef8c24c67e826b801b189b763742b44b8af2334443)`

**Comment by daniel-savu** [comment]
I'm not familiar with the CLI version of polkadot-js, but these are the extrinsics you need to set up USDT and three lending markets on Kintnet (KSM, KBTC, USDT):
- addMarket (KBTC): https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fapi-dev-kintsugi.interlay.io%2Fparachain#/extrinsics/decode/0x02006100000bd04e090010eb0900400d030020a107000000ee042cfc430f0000000000000000a861000000000000000000000000000000000000000000c52ebca2b10000000000000000000000b2d3595bf0060000000000000000a0bb0d000100943577000000000000000000000000009435770000000000000000000000000201000000
- activateMarket (KBTC): https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fapi-dev-kintsugi.interlay.io%2Fparachain#/extrinsics/decode/0x02006101000b
- addMarket (KSM): https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fapi-dev-kintsugi.interlay.io%2Fparachain#/extrinsics/decode/0x02006100000a603d0800d04e0900400d030020a107000000ee042cfc430f0000000000000000a8610000000000000000000000000000000000000000004f8c34e81402000000000000000000002876e1158d050000000000000000a0bb0d00010000434fd7946a0000000000000000000000434fd7946a0000000000000000000202000000
- activateMarket (KSM): https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fapi-dev-kintsugi.interlay.io%2Fparachain#/extrinsics/decode/0x02006101000a
- assetRegistry.registerAsset (USDT): https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fapi-dev-kintsugi.interlay.io%2Fparachain#/extrinsics/decode/0x0200180006000000285465746865722055534410555344540000000000000000000000000000000000b067b500000000000000000000000000187465746865720101000000
- addMarket (USDT): https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fapi-dev-kintsugi.interlay.io%2Fparachain#/extrinsics/decode/0x02006100010100000010eb090070d50a00400d030020a107000000ee042cfc430f0000000000000000a8610000000000000000000000000000000000000000004f8c34e81402000000000000000000002876e1158d050000000000000000a0bb0d00010000434fd7946a0000000000000000000000434fd7946a0000000000000000000203000000
- activateMarket (USDT): https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fapi-dev-kintsugi.interlay.io%2Fparachain#/extrinsics/decode/0x020061010101000000

We don't have fine-tuned parameters for the lending markets on interlay-testnet so you can just change the currencies in the calls above (e.g. replace KBTC with IBTC above, etc).

File: scripts/init-testnet-parachain.sh [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        "liquidateIncentiveReservedFactor": 25000,
+        "rateModel": {
+          "Jump": {
+            "baseRate": 0,
+            "jumpRate": "15000000000000000",
+            "fullRate": "40000000000000000",
+            "jumpUtilization": 900000
+          }
+        },
+        "state": "Active",
```
Comment (daniel-savu) [inline scripts/init-testnet-parachain.sh:108]: the state has to be `Pending` when a market is first added. To activate after adding, an `activateMarket` extrinsic has to be called

File: scripts/init-testnet-parachain.sh [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+      2121 \
+      "{ \"genesisHead\":\"${GENESIS_FILENAME?}\", \"validationCode\":\"${WASM_FILENAME?}\", \"parachain\": true }"
+
+  # forceLease(para, leaser, amount, periodBegin, periodCount)
+  polkadot-js-api --ws $RELAYCHAIN_WSS --sudo --seed "//Alice" \
+      tx.slots.forceLease \
+      2121 \
+      5CiPPseXPECbkjWCa6MnjNokrgYjMqmKndv2rSnekmSK2DjL \
+      0 \
+      1 \
```
Comment (sander2) [inline scripts/init-testnet-parachain.sh:51]: should be 0

File: scripts/init-testnet-parachain.sh [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+      1000
```
Comment (sander2) [inline scripts/init-testnet-parachain.sh:52]: use 100000

**Comment by nud3l** [comment]
@gregdhill @sander2 should we close this?

**Comment by gregdhill** [comment]
Succeeded by https://github.com/interlay/interbtc-api/pull/554

**Pull request closed without merging**